pkg/pkgupdates/pkgupdates.jsx

```diff
@@ -0,0 +1,532 @@
+var $ = require('jquery');
+var cockpit = require('cockpit');
+var React = require('react');
+var moment = require('moment');
+var Tooltip = require('cockpit-components-tooltip.jsx').Tooltip;
+
+const _ = cockpit.gettext;
+
+// 'available' heading is built dynamically
+const STATE_HEADINGS = {
+    'loading': _('Loading available updates, please wait...'),
+    'locked': _('Some other program is currently using the package manager, please wait...'),
+    'uptodate': _('No updates available, system is up to date'),
+    'applying': _('Applying updates'),
+    'updateError': _('Applying updates failed'),
+    'loadError': _('Loading available updates failed'),
+}
+
+// see https://github.com/hughsie/PackageKit/blob/master/lib/packagekit-glib2/pk-enum.h
+const PK_EXIT_ENUM_SUCCESS = 1;
+const PK_EXIT_ENUM_FAILED = 2;
+const PK_ROLE_ENUM_REFRESH_CACHE = 13;
+const PK_ROLE_ENUM_UPDATE_PACKAGES = 22;
+const PK_INFO_ENUM_SECURITY = 8;
+const PK_STATUS_ENUM_WAIT = 1;
+const PK_STATUS_ENUM_UPDATE = 10;
+const PK_STATUS_ENUM_WAITING_FOR_LOCK = 1;
+
+const PK_STATUS_STRINGS = {
+    9: _('Downloading'),
+    10: _('Updating'),
+    11: _('Cleaning up'),
+    14: _('Verifying'),
+}
+
+var dbus_pk = cockpit.dbus('org.freedesktop.PackageKit', {superuser: 'try'});
+var packageSummaries = {};
+
+function pkTransaction() {
+    var dfd = cockpit.defer();
+
+    dbus_pk.call('/org/freedesktop/PackageKit', 'org.freedesktop.PackageKit', 'CreateTransaction', [], {timeout: 5000})
+        .done(result => {
+            var transProxy = dbus_pk.proxy('org.freedesktop.PackageKit.Transaction', result[0]);
+            transProxy.wait(() => {dfd.resolve(transProxy)});
+        })
+        .fail(ex => {dfd.reject(ex)});
+
+    return dfd.promise();
+}
+
+// parse CVEs from an arbitrary text (changelog) and return URL array
+function parseCVEs(text) {
+    if (!text)
+        return [];
+
+    var cves = text.match(/CVE-\d{4}-\d+/g);
+    if (!cves)
+        return [];
+    return cves.map(n => 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=' + n);
+}
+
+function deduplicate(list) {
+    var d = { };
+    list.forEach(i => {if (i) d[i] = true});
+    var result = Object.keys(d);
+    result.sort();
+    return result;
+}
+
+function commaJoin(list) {
+    return list.reduce((prev, cur) => [prev, ', ', cur])
+}
+
+function HeaderBar(props) {
+    var num_updates = Object.keys(props.updates).length;
+    var num_security = 0;
+    var state;
+    if (props.state == 'available') {
+        state = cockpit.ngettext('$0 update', '$0 updates', num_updates);
+        for (var u in props.updates)
+            if (props.updates[u].security)
+                ++num_security;
+        if (num_security > 0)
+            state += cockpit.ngettext(', including $1 security fix', ', including $1 security fixes',  num_security);
```

Comment:
Since there is no multi-dimensional ngettext (the combinatorial explosion would kill translators), we need to keep the two apart somehow. We could use parentheses or a slash as a separator instead of a comma in the C string, but this is not really a sentence but an enumeration.

if* there is a language that doesn't have comma enumeration and the order of "a, including b" is reversed with no logical possibility of using the English order, then that translation should reformat it as appropriate - it could say  "$0 qu'moH" and " ($1 niH ga moH)", anything which keeps the security number second.